nwb_conversion_tools/utils/spike_interface.py

```diff
@@ -331,27 +332,65 @@ def add_electrodes(
 
     assert all([isinstance(x, dict) and set(x.keys()) == set(['name', 'description', 'data'])
                 and isinstance(x['data'], list) for x in metadata['Ecephys']['Electrodes']]), \
         "Expected metadata['Ecephys']['Electrodes'] to be a list of dictionaries!"
     assert all([x['name'] != 'group' for x in metadata['Ecephys']['Electrodes']]), \
-        "Passing metadata field 'group' is depricated; pass group_name instead!"
+        "Passing metadata field 'group' is deprecated; pass group_name instead!"
 
     if nwbfile.electrodes is None:
         nwb_elec_ids = []
     else:
         nwb_elec_ids = nwbfile.electrodes.id.data[:]
 
-    for metadata_column in metadata['Ecephys']['Electrodes']:
-        if (nwbfile.electrodes is None or metadata_column['name'] not in nwbfile.electrodes.colnames) \
-                and metadata_column['name'] not in defaults:
-            nwbfile.add_electrode_column(
-                name=str(metadata_column['name']),
-                description=str(metadata_column['description'])
-            )
+    elec_columns = {}  # name: description
+    property_names = set()
+    for chan_id in recording.get_channel_ids():
+        for i in recording.get_channel_property_names(channel_id=chan_id):
+            property_names.add(i)
+    for prop in property_names:
+        if prop not in ['gain', 'offset', 'location', 'name'] + list(exclude):
+            # property 'gain' should not be in the NWB electrodes_table
+            # property 'brain_area' of RX channels corresponds to 'location' of NWB electrodes
+            # property 'offset' should not be in the NWB electrodes_table as not officially supported by schema v2.2.5
+
+            data = []
+            for chan_id in recording.get_channel_ids():
+                if prop in recording.get_channel_property_names(channel_id=chan_id):
+                    data.append(recording.get_channel_property(channel_id=chan_id, property_name=prop))
+                else:
+                    if len(data) > 0 and isinstance(data[-1], str):
+                        data.append('')
+                    elif len(data) > 0:
+                        data.append(np.nan)
+            prop = 'location' if prop == 'brain_area' else prop
+            prop = 'group_name' if prop == 'group' else prop
+            elec_columns[prop] = dict(description=prop, data=data)
+
+    for x in metadata['Ecephys']['Electrodes']:
+        name = x.pop('name')
+        if name not in elec_columns:
+            elec_columns[name] = dict(description=name)
+        if 'description' in x:
+            elec_columns[name]['description'] = x['description']
+        if 'data' in x:
+            assert len(x['data']) == recording.get_num_channels()
+            elec_columns[name]['data'] = x['data']
+
+    for name, dat in elec_columns.items():
+        if 'data' not in dat and name not in defaults:
+            raise ValueError(f"no data for electrode column {name}")
```

Comment:
Given we have complete control over the content and structure of the `elec_columns `object, I don't think we need to `raise` this. I'll keep thinking through it to be sure.

Reply:
see comment above. I'd like it to be set-able in the properties of the extractor